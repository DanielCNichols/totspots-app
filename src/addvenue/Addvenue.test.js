import React from 'react';
import ReactDOM from 'react-dom'
import AddVenue from './AddVenue.js'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'



it('renders AddVenue without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <AddVenue />
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div)
})

it('renders AddVenue as expected', () => {
  const tree = renderer
  .create(
  <MemoryRouter>
    <AddVenue name="AddVenue"/>
  </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})