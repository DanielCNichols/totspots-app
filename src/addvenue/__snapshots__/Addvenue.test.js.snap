// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddVenue as expected 1`] = `
<section
  className="addVenue"
>
  <header>
    <h2>
      Add A Venue
    </h2>
  </header>
  <div
    role="alert"
  />
  <form
    className="add_form"
    onSubmit={[Function]}
  >
    <fieldset>
      +
      <legend>
        Venue Information
        <div
          className="address"
        >
          <label
            className="add_form_label"
            htmlFor="venue"
          >
            Venue/Event Name
            <input
              className="add_form_text"
              id="Venue"
              name="venue"
              placeholder="ex: Bull McCabe's"
              required={true}
              type="text"
            />
          </label>
          <label
            htmlFor="address"
          >
            Street Address
            <input
              className="add_form_text"
              name="address"
              placeholder="123 Main St."
              required={true}
              type="text"
            />
          </label>
          <label
            htmlFor="city"
          >
            City
            <input
              className="add_form_text"
              name="city"
              placeholder="ex: Durham"
              required={true}
              type="text"
            />
          </label>
          <label
            htmlFor="state"
          >
            State
            <input
              className="add_form_text"
              maxLength="2"
              name="state"
              placeholder="ex: NC"
              required={true}
              type="text"
            />
          </label>
          <label
            htmlFor="zipcode"
          >
            Zipcode
            <input
              className="add_form_text"
              name="zipcode"
              placeholder="ex: 27705"
              required={true}
              type="text"
            />
          </label>
          <label
            htmlFor="url"
          >
            Website
            <input
              className="add_form_text"
              name="url"
              placeholder="www.reddit.com"
              required={true}
              type="url"
            />
          </label>
          <label
            htmlFor="phone"
          >
            Phone
            <input
              className="add_form_text"
              name="phone"
              placeholder="(555) 245-3456"
              required={true}
              type="text"
            />
          </label>
          <div
            className="add_form_selects"
          >
            <label
              htmlFor="type"
            >
              Type of venue/event
              <select
                className="add_form_select"
                id="type"
                name="type"
                required={true}
              >
                <option
                  value=""
                >
                  Please select
                </option>
                <option
                  value="Restaurant"
                >
                  Restaurant
                </option>
                <option
                  value="Bar"
                >
                  Brewery/Bar
                </option>
                <option
                  value="Educational"
                >
                  Educational
                </option>
                <option
                  value="music"
                >
                  Music
                </option>
                <option
                  value="outdoor"
                >
                  Outdoor
                </option>
              </select>
            </label>
          </div>
        </div>
      </legend>
    </fieldset>
    <fieldset>
      <legend>
        Your review
        <label
          htmlFor="price"
          required={true}
        >
          Price
          <select
            className="add_form_select"
            id="price"
            name="price"
            required={true}
          >
            <option
              value=""
            >
              Please select
            </option>
            <option
              value="1"
            >
              $
            </option>
            <option
              value="2"
            >
              $$
            </option>
            <option
              value="3"
            >
              $$$
            </option>
            <option
              value="4"
            >
              $$$$
            </option>
            <option
              value="5"
            >
              $$$$$
            </option>
          </select>
        </label>
        <label
          htmlFor="volume"
        >
          Describe the volume Level
          <select
            className="add_form_select"
            id="volume"
            name="volume"
            required={true}
          >
            <option
              value=""
            >
              Please select
            </option>
            <option
              value="1"
            >
              Library
            </option>
            <option
              value="2"
            >
              Coffee Shop
            </option>
            <option
              value="3"
            >
              Restaurant
            </option>
            <option
              value="4"
            >
              Bar/Brewery
            </option>
            <option
              value="5"
            >
              Concert
            </option>
          </select>
        </label>
        <label
          htmlFor="rating"
        >
          Overall rating
          <select
            className="add_form_select"
            id="rating"
            name="rating"
            required={true}
          >
            <option
              value=""
            >
              Please select
            </option>
            <option
              value="1"
            >
              ★
            </option>
            <option
              value="2"
            >
              ★ ★
            </option>
            <option
              value="3"
            >
              ★ ★ ★
            </option>
            <option
              value="4"
            >
              ★ ★ ★ ★
            </option>
            <option
              value="5"
            >
              ★ ★ ★ ★ ★
            </option>
          </select>
        </label>
      </legend>
    </fieldset>
    <fieldset>
      <legend>
        Features
        <div
          className="amenities"
        >
          <label
            htmlFor="StrollerAccessible"
          >
            Stroller Accessible
            <input
              name="amenities"
              onChange={[Function]}
              type="checkbox"
              value="1"
            />
          </label>
          <label
            htmlFor="PlayArea"
          >
            Play Area
            <input
              name="PlayArea"
              onChange={[Function]}
              type="checkbox"
              value="2"
            />
          </label>
          <label
            htmlFor="Changingtable"
          >
             
            Changing Table
            <input
              name="amenities"
              onChange={[Function]}
              type="checkbox"
              value="3"
            />
          </label>
          <label
            htmlFor="Dogs"
          >
            Dogs Welcome
            <input
              name="amenities"
              onChange={[Function]}
              type="checkbox"
              value="4"
            />
          </label>
          <label
            htmlFor="Fastcheckout"
          >
             
            Fast Checkout
            <input
              name="amenities"
              onChange={[Function]}
              type="checkbox"
              value="5"
            />
          </label>
          <label
            htmlFor="KidsNight"
          >
             
            Kids Night Deals
            <input
              name="amenities"
              onChange={[Function]}
              type="checkbox"
              value="6"
            />
          </label>
          <label
            htmlFor="OutdoorSeating"
          >
             
            Outdoor Seating
            <input
              name="amenities"
              onChange={[Function]}
              type="checkbox"
              value="7"
            />
          </label>
        </div>
        <div
          className="add_form_review"
        >
          <label
            htmlFor="review"
          >
            Tell us about your visit
            <textarea
              name="content"
              placeholder="Tell us about your visit"
              type="textArea"
            />
          </label>
        </div>
      </legend>
    </fieldset>
    <div
      className="add_form_controls"
    >
      <button
        className="cancel"
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="add"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</section>
`;
